apps/backend/src/infrastructure/repositories/users/accounts.ts

```diff
@@ -75,10 +97,26 @@ const getAll = async (): Promise<Account[]> => {
   return mapRows(result.rows)
 }
 
+export const getTopAccountsWithinPeriod = async (
+  limit: number = 10,
+  type: InteractionType = InteractionType.Upload,
+  fromDate: Date,
+  toDate: Date,
+): Promise<AccountWithTotalSize[]> => {
+  const db = await getDatabase()
+
+  const result = await db.query<DBAccountWithTotalSize>(
+    'SELECT a.*, SUM(i.size) as total_size FROM accounts a INNER JOIN interactions i ON a.id = i.account_id WHERE i.type = $1, i.created_at > $2 and i.created_at < $3 GROUP BY a.id ORDER BY total_size DESC LIMIT $4',
```

Comment:
In the `WHERE` clause you are using a comma instead of `AND`.
```suggestion
    'SELECT a.*, SUM(i.size) as total_size FROM accounts a INNER JOIN interactions i ON a.id = i.account_id WHERE i.type = $1 AND i.created_at > $2 AND i.created_at < $3 GROUP BY a.id ORDER BY total_size DESC LIMIT $4',
```